Use .rakumod extensions instead of .pm6

lizmat opened the pull request.

vrurg commented:
Hail to the brave ones! :)

There are various `make*` scripts in tools/build which are writing directly into .pm6 under the src/core.c/.

lizmat commented:
This PR went stale.  Cherry-picked the additional commits and renamed the files again, and squashed this into https://github.com/rakudo/rakudo/commit/81d2aa6fc2
